go-controller/pkg/node/node.go

```diff
@@ -595,6 +604,79 @@ func (n *OvnNode) WatchEndpoints() error {
 	return err
 }
 
+func exGatewayPodsAnnotationsChanged(oldNs, newNs *kapi.Namespace) bool {
+	// In reality we only care about exgw pod deletions, however since the list of IPs is not expected to change
+	// that often, let's check for *any* changes to this annotation compared to its previous state and trigger
+	// the logic for checking if we need to delete any conntrack entries
+	return oldNs.Annotations[util.ExternalGatewayPodIPsAnnotation] != newNs.Annotations[util.ExternalGatewayPodIPsAnnotation]
+}
+
+func (n *OvnNode) checkAndDeleteStaleConntrackEntries() {
+	namespaces, err := n.watchFactory.GetNamespaces()
+	if err != nil {
+		klog.Errorf("Unable to get pods from informer: %v", err)
+	}
+	for _, namespace := range namespaces {
```

Comment:
so this will iterate all namespaces, and then in each namespace, serially arp for each GW IP. Would it be better to parallelize at least the ARPs? Maybe it's overkill...

Reply:
probably yes I was under the impression not many namespaces will have the exgw related annotations, but I could be wrong.

Reply:
done parallelized this into routines within loop